cmd/reaper/types_watcher.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/boskos/common"
+)
+
+type ResourceTypesWatcher struct {
+	config        string
+	providedTypes []string
```

Comment:
nit: if you need this as a `sets.String` on each load, might as well keep that around instead of re-allocating every time